src/typing/type_inference_js.ml

```diff
@@ -930,8 +930,12 @@ and convert_qualification ?(lookup_mode=ForType) cx reason_prefix
 )
 
 and mk_rest cx = function
-  | ArrT(_, t, []) -> RestT t
+  | ArrT (_, t, []) -> RestT t
   | AnyT _ as t -> RestT t
+  | ArrT (r, t, _) ->
+      let msg = "rest parameters should be an array type, got a tuple type instead" in
+      Flow_js.add_warning cx [r,msg];
+      RestT t
```

Comment:
A pattern exhibited elsewhere is to propagate `AnyT` in the event of an error (presumably, so errors don't just pile up). Would it be better to error and return `RestT AnyT` here instead of plowing ahead with the combined element type of the union?